File: Contracts/Model/Enterprise/Response/QuoteResponse.cs
namespace Contracts.Model.Enterprise.Response;

public class QuoteResponse : BaseResponse
{
    public int Id { get; set; }
    public string CustomerName { get; set; }
    public string ContactNumber { get; set; }
    public string? ExtraInfo { get; set; }
    public string? Email { get; set; }
    public string EventName { get; set; }
    public DateTimeOffset EventDate { get; set; }
    public string Location { get; set; }
    public ICollection<QuoteItemResponse> QuoteItems { get; set; }
    public string? UserId { get; set; }
}

File: ClientApp/src/app/core/interfaces/Enterprise/Response/quoteResponse.ts
import { PurchaseDetailResponse } from "../../Product/UserInteract/Response/purchaseDetailResponse";
import { BaseResponse } from "../../baseResponse";

export interface QuoteResponse extends BaseResponse {
    id: string;
    customerName: string;
    contactNumber: string;
    extraInfo: string | null;
    email: string | null;
    eventName: string;
    eventDate: string;
    location: string;
    quoteItems: PurchaseDetailResponse[];
    userId: string | null;
}